modules/db/src/test/java/org/mule/module/db/internal/domain/database/GenericDbConfigFactoryTestCase.java

```diff
@@ -0,0 +1,110 @@
+{
+
+    private final GenericDbConfigFactory genericDbConfigFactory = new GenericDbConfigFactory();
+    private final Map<QName, Object> annotations = mock(Map.class);
+    private final DataSource dataSource = mock(DataSource.class);
+    private final List<DbType> customDataTypes = new ArrayList<>();
+    private final MuleContext muleContext = mock(MuleContext.class);
+    private final MuleRegistry muleRegistry = mock(MuleRegistry.class);
+    private final RetryPolicyTemplate globalRetryPolicyTemplate = mock(RetryPolicyTemplate.class);
+    private final NoRetryPolicyTemplate noRetryPolicyTemplate = mock(NoRetryPolicyTemplate.class);
+    private final RetryPolicyTemplate specificRetryPolicyTemplate = mock(RetryPolicyTemplate.class);
+    private DbConfig dbConfig = null;
+    private TransactionalDbConnectionFactory transactionalDbConnectionFactory = null;
+    private RetryConnectionFactory retryConnectionFactory = null;
+
+    @Before
+    public void setUp() throws Exception
+    {
+        genericDbConfigFactory.setMuleContext(muleContext);
+        genericDbConfigFactory.setCustomDataTypes(customDataTypes);
+        when(muleContext.getRegistry()).thenReturn(muleRegistry);
+        when(globalRetryPolicyTemplate.isSynchronous()).thenReturn(true);
+        when(specificRetryPolicyTemplate.isSynchronous()).thenReturn(true);
+    }
+
+    @Test
+    public void testOnlyExistsGlobalRetryPolicyTemplate() throws Exception
+    {
+        when(muleRegistry.lookupObject(OBJECT_DEFAULT_RETRY_POLICY_TEMPLATE)).thenReturn(globalRetryPolicyTemplate);
+        dbConfig = genericDbConfigFactory.create("name", annotations, dataSource);
+        assertThat(dbConfig.getConnectionFactory(), instanceOf(TransactionalDbConnectionFactory.class));
+        transactionalDbConnectionFactory = (TransactionalDbConnectionFactory) dbConfig.getConnectionFactory();
+        assertThat(transactionalDbConnectionFactory.getConnectionFactory(), instanceOf(RetryConnectionFactory.class));
+        retryConnectionFactory = (RetryConnectionFactory) transactionalDbConnectionFactory.getConnectionFactory();
+        assertEquals(retryConnectionFactory.getRetryPolicyTemplate(), globalRetryPolicyTemplate);
+    }
+
+    @Test
+    public void testOnlyExistsSpecificRetryPolicyTemplate() throws Exception
+    {
+        genericDbConfigFactory.setRetryPolicyTemplate(specificRetryPolicyTemplate);
+        dbConfig = genericDbConfigFactory.create("name", annotations, dataSource);
+        assertThat(dbConfig.getConnectionFactory(), instanceOf(TransactionalDbConnectionFactory.class));
+        transactionalDbConnectionFactory = (TransactionalDbConnectionFactory) dbConfig.getConnectionFactory();
+        assertThat(transactionalDbConnectionFactory.getConnectionFactory(), instanceOf(RetryConnectionFactory.class));
+        retryConnectionFactory = (RetryConnectionFactory) transactionalDbConnectionFactory.getConnectionFactory();
```

Comment:
can you use a mock for globalRetryPolicyTemplate to avoid adding the new getters?